src/julia.h

```diff
@@ -1940,6 +1940,7 @@ typedef struct _jl_task_t {
     void *stkbuf; // malloc'd memory (either copybuf or stack)
     size_t bufsz; // actual sizeof stkbuf
     uint64_t inference_start_time; // time when inference started
+    unsigned int timing_inference; // whether inference is currently timing
```

Comment:
Do we need to use another 4 bytes for this?

Reply:
If we believe that inference and codegen aren't going to exceed 65536 recursions, we can drop all of the recursion trackers to 2 bytes.

Reply:
But yes I think the issue was that I wasn't initializing the inference start time to 0, so junk times ended up being recorded. I'll adjust it to rely on the reentrant inference tracker again.